### ISSUE TITLE
feature: SBOM needed to clomonitor tests

### ISSUE THREAD

**Issue by clubanderson** [kubestellar/kubestellar#2613]
### Feature Description

clomonitor requires a valid SBOM is present in a repo to improve the repo score

**SOFTWARE BILL OF MATERIALS (SBOM)**
ID: sbom

List of components in a piece of software, including licenses, versions, etc.

This check passes if:

The latest release on Github includes an asset which name contains sbom. Regexps used:
"(?i)sbom"
The repository’s README file contains a SBOM section that explains where they are published to, format used, etc. Regexps used to locate the title header:
"(?im)^#+.*sbom.*$"
"(?im)^#+.*software bill of materials.*$"
"(?im)^sbom$"
"(?im)^software bill of materials$"

### Proposed Solution

produce SBOM with our CICD

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by jaydee029** [comment]
Hey @clubanderson, I'd like to work on this,  have done something similar for wasmedge before.

**Comment by clubanderson** [comment]
@jaydee029 yes. Please take the issue. Thank you for you contribution in advance.

**Comment by MikeSpreitzer** [comment]
If we remove the binary archives from the releases (see #2653), would that make this moot?

**Comment by jaydee029** [comment]
> If we remove the binary archives from the releases (see #2653), would that make this moot?

So I'm not sure, but I think SBOM here is concerned with everything you used to build this particular software, If I go into the release section on github I find this

![image](https://github.com/user-attachments/assets/0a20a67e-1a18-47e5-ac7a-f78df20c0177)

Since we do have packages/binaries of some sort or the other in our release, SBOM would be required.
I might be wrong.

**Comment by MikeSpreitzer** [comment]
If the binary archives go, would that make clomonitor stop caring about an SBOM, or would it still be looking for an SBOM covering the source archives?

**Comment by jaydee029** [comment]
Please check the discussion [here](https://github.com/kubestellar/kubestellar/issues/2653#issuecomment-2576959955)

**Comment by MikeSpreitzer** [comment]
I just created a test release, 0.26.0-alpha.5, to test the intended fix for this Issue, PR #2718 . The test release did _not_ get an SBOM in its assets.

**Comment by MikeSpreitzer** [comment]
@jaydee029

**Comment by jaydee029** [comment]
Sure ,I'll look into this soon, I've been busy for a last few days, sorry for the delay

**Comment by jaydee029** [comment]
@MikeSpreitzer its always hard to test these releases locally, but I have added a `syft binary` in the workflow as recommended by a  blog released by goreleaser (3 years ago) , although they don't mention anything about this in their documentation. This should work.

**Comment by MikeSpreitzer** [comment]
Tested in the creation of the 0.26.0-rc.2 release. Still, no SBOM created.
